packages/store/addon/-private/system/core-store.ts

```diff
@@ -1403,6 +1403,10 @@ abstract class CoreStore extends Service {
         groups = [snapshots];
       }
 
+      // we use var here because babel transpiles let
+      // in a manner that causes a mega-bad perf scenario here
+      // when targets no longer include IE11 we can drop this.
+      /* eslint-disable no-var */
```

Comment:
Huh. Out of curiosity, is this still true? The Babel playground just compiles it to `for (var i = 0, ...)`:

<img width="1234" alt="Screen Shot 2021-04-13 at 09 22 46" src="https://user-images.githubusercontent.com/2403023/114578071-d1b06780-9c39-11eb-89a6-49576eff8bb7.png">

Reply:
I checked it and it was still true which is why I added the comment. Though the playground not compiling it to the closed over functions means there may be something for us to tweak that will allow us to remove this.

That said, on your left there note how you didn't change the `var` definitions in the for loops themselves. IIRC only two of these actually have to be vars for us to not transpile to crazy land (`ids` and `internalModel` or maybe it was `group`)

Reply:
I checked the playground and if you convert all the vars to let and transpile to IE11 (bc modern doesn't transpile scope things anymore) you do hit the pathological perf case with functions redefined on every iteration that have their own loops redefined with functions etc.